addons/l10n_it_edi/models/ir_mail_server.py

```diff
@@ -123,35 +146,47 @@ def _attachment_invoice(self, msg_txt):
                     _logger.info('New E-invoice in zip file: %s', attachment.fname)
                     self._create_invoice_from_mail_with_zip(attachment, from_address)
 
-    def _create_invoice_from_mail(self, att_content, att_name, from_address):
-        if self.env['account.move'].search([('l10n_it_einvoice_name', '=', att_name)], limit=1):
-            # invoice already exist
+    def _create_invoice_from_mail(self, att_content_data, att_name, from_address):
+
+        invoices = self.env["account.move"]
+
+        # Check if we already imported the email as an attachment
+        existing = self.env['ir.attachment'].search([("name", "=", att_name), ("type", "=", "binary")])
```

Comment:
@lordkrandel Would it hurt if we check if the res_model of the attachment is 'account.invoice'?  (and avoid the type binary)